lib/matplotlib/tests/test_contour.py

```diff
@@ -693,3 +693,26 @@ def test_contour_remove():
     assert ax.get_children() != orig_children
     cs.remove()
     assert ax.get_children() == orig_children
+
+
+def test_contour_z_kwargonly():
+    # Smoke test for GH#24743
+    # Passing only kwargs
+    plt.contour(Z=np.random.rand(30, 30))
+
+
+def test_contour_xyz_kwargsonly():
+    plt.contour(
+        X=np.random.rand(30, 30),
+        Y=np.random.rand(30, 30),
+        Z=np.random.rand(30, 30)
+    )
```

Comment:
Note, I also think we don't need a large random dataset:
```suggestion
    data = [[0, 1], [1, 0]]
    plt.contour(Z=data)
    plt.contour(X=data, Y=data, Z=data)
```

Reply:
Fair.

Reply:
Done.